elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala

```diff
@@ -368,4 +370,64 @@ trait QueryDsl extends DslCommons with SortDsl {
       _dis_max -> innerMap
     )
   }
+
+  case class HighlightRoot(queryRoot: QueryRoot, highlight: Highlight)
```

Comment:
maybe I'm missing something -- can we just have HighlightRoot extend RootObject rather than creating a separate `QueryRootBase`?